lib/archethic/contracts/interpreter/library/common/http_impl.ex

```diff
@@ -192,14 +206,28 @@ defmodule Archethic.Contracts.Interpreter.Library.Common.HttpImpl do
     end
   end
 
+  defp validate_request(url, method, headers, body)
+       when is_binary(url) and is_binary(method) and is_map(headers) and
+              (is_binary(body) or is_nil(body)) do
```

Comment:
Suggestion: ideally I would prefer a `with` statement with appropriate private function to return ok/error tuple, as it would result in more simple and clean function body.